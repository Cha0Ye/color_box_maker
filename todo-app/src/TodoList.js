import React, { Component } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
// import './App.css';
import uuid from 'uuid/v4';

class TodoList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            todolist: [],
        }
        this.create = this.create.bind(this);
        this.delete = this.delete.bind(this);
    }

    create(obj) {
        const newtodo = { ...obj, id: uuid() };
        this.setState( state => ({ todolist:[...state.todolist, newtodo] }));
    }

    delete(n) {
        const newTodo = this.state.todolist.filter((ele, idx) => idx !== n);
        this.setState({todolist: newTodo});
    }

    renderTodo(){
        this.state.todolist.map((ele, idx) => <Todo id={ele.id} text={ele} triggerDelete={() => this.delete(idx)}/>)
    }

    render() {
        return (
            <div className="TodoList">
                <NewTodoForm triggerCreate={ this.create }/>
                {this.renderTodo()}
            </div>
        );
    }
}

export default TodoList;
