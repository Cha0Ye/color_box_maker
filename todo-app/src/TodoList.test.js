import React from 'react';
import TodoList from './TodoList';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

let wrapper;

beforeEach(() => {
    wrapper = mount(<TodoList />);
})

//smoke test
it('renders without crashing', () => {
  const wrapper = shallow(<TodoList />);
});

//snapshot test
it('matches snapshot', () => {

    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});
