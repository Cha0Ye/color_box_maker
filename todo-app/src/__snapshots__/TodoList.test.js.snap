// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TodoList>
  <div
    className="TodoList"
  >
    <NewTodoForm
      triggerCreate={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="text"
        >
          Todo: 
        </label>
        <input
          id="text"
          name="text"
          onChange={[Function]}
          value=""
        />
        <button>
          Create
        </button>
      </form>
    </NewTodoForm>
  </div>
</TodoList>
`;
